osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,189 @@
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/tables/system/windows/userassist.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kFullRegPath =
+    "\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\UserAssist";
+
+// Decode ROT13 sub key value
+std::string rot_decode(std::string& value_key_reg) {
+  std::string decoded_value_key;
+
+  for (std::size_t i = 0; i < value_key_reg.size(); i++) {
+    if (isalpha(value_key_reg[i])) {
+      if (value_key_reg[i] >= 'a' && value_key_reg[i] <= 'm') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'm' && value_key_reg[i] <= 'z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      } else if (value_key_reg[i] >= 'A' && value_key_reg[i] <= 'M') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'M' && value_key_reg[i] <= 'Z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      }
+    } else {
+      decoded_value_key.append(1, value_key_reg[i]);
+    }
+  }
+  return decoded_value_key;
+}
+
+// Get last exeution time
+auto last_execute(std::string& assist_data) {
+  if (assist_data.length() <= 136) {
+    LOG(WARNING) << "Userassist last execute Timestamp format is incorrect";
+    return std::string();
+  }
+  std::string last_run_string = assist_data.substr(120, 16);
+
+  // If timestamp is zero dont convert to UNIX Time
+  if (last_run_string == "0000000000000000") {
+    return std::string();
+  } else {
+    // swap endianess
+    std::reverse(last_run_string.begin(), last_run_string.end());
+
+    for (std::size_t i = 0; i < last_run_string.length(); i += 2) {
+      char temp = last_run_string[i];
+      last_run_string[i] = last_run_string[i + 1];
+      last_run_string[i + 1] = temp;
+    }
+
+    // Convert Windows FILETIME to UNIX Time
+    // unsigned long long last_run = std::stoull(last_run_string.c_str(), 0,
+    // 16);
+    unsigned long long last_run =
+        tryTo<unsigned long long>(last_run_string, 16).takeOr(0ull);
+    if (last_run == 0ull) {
+      LOG(WARNING) << "Failed to convert FILETIME to UNIX time.";
+      return std::string();
+    }
+    last_run = (last_run / 10000000) - 11644473600;
+
+    std::time_t last_run_time = last_run;
+
+    struct tm tm;
+    gmtime_s(&tm, &last_run_time);
+
+    auto time_str = platformAsctime(&tm);
+    return time_str;
+  }
+}
+
+// Get execution count
+int execution_num(std::string& assist_data) {
```

Comment:
nit: looks like we're using `camlCase` down below for naming `genUserAssist`. I believe we prefer `camlCase`ing for all of the functions, do you mind changing your function signatures to make everything uniform?

Reply:
converted functions to camelcase